index.js

```diff
@@ -18,78 +17,57 @@ module.exports = function () {
 		return Promise.resolve(fullname);
 	}
 
-	return pify(npmconf.load, Promise)().then(function (conf) {
+	return pify(npmconf.load)().then(conf => {
 		fullname = conf.get('init.author.name');
 
 		if (!fullname) {
-			return fallback();
+			return fallback().then(() => fullname);
 		}
 
 		return fullname;
-	}).catch(fallback).catch(function () {});
+	}).catch(fallback).then(() => fullname).catch(() => {});
 };
 
 function fallback() {
 	if (process.platform === 'darwin') {
-		return exec('id -P')
-			.then(function (stdout) {
-				fullname = stdout.trim().split(':')[7];
-
-				if (!fullname) {
-					throw new Error();
-				}
-
-				return fullname;
-			})
-			.catch(function () {
-				// `id -P` should never fail as far as I know, but just in case:
-				return exec('osascript -e "long user name of (system info)"')
-					.then(function (stdout) {
-						fullname = stdout.trim();
-
-						return fullname;
-					});
+		return passwdUser(process.getuid())
+			.then(user => fullname = user.fullname)
+			.catch(() => {
+				return execa('osascript', ['-e', '"long user name of (system info)"'])
+					.then(res => fullname = res.stdout);
 			});
 	}
 
 	if (process.platform === 'win32') {
 		// try git first since fullname is usually not set by default in the system on Windows 7+
-		return exec('git config --global user.name')
-			.then(function (stdout) {
-				fullname = stdout.trim();
+		return execa('git', ['config', '--global', 'user.name'])
+			.then(res => {
+				fullname = res.stdout;
 
 				if (!fullname) {
 					throw new Error();
 				}
-
-				return fullname;
 			})
-			.catch(function () {
-				return exec('wmic useraccount where name="%username%" get fullname')
-					.then(function (stdout) {
-						fullname = stdout.replace('FullName', '').trim();
-
-						return fullname;
-					});
+			.catch(() => {
+				return execa('wmic', ['useraccount', 'where', 'name="%username%"', 'get', 'fullname'])
+					.then(res => fullname = res.stdout.replace('FullName', ''));
 			});
 	}
 
-	return exec('getent passwd $(whoami)')
-		.then(function (stdout) {
-			fullname = (stdout.trim().split(':')[4] || '').replace(/,.*/, '');
-
-			if (!fullname) {
-				throw new Error();
-			}
+	return passwdUser(process.getuid())
+		.then(user => fullname = user.fullname)
+		.catch(() => {
+			return execa('getent', ['passwd', '$(whoami)'])
```

Comment:
I think this is the problem. You can't do `$(whoami)` as `$()` is a shell thing and here we're calling the `getent` binary directly. Easiest would be to just revert to use `execa.shell` here, otherwise you'd need to do a `whoami` call before this and that would be more verbose and probably slower than `execa.shell`.